scripts/ai/strategies/AIDriveStrategyFindBales.lua

```diff
@@ -224,13 +224,15 @@ end
 function AIDriveStrategyFindBales:findBales()
     local balesFound, baleWithWrongWrapType = {}, false
     for _, object in pairs(g_baleToCollectManager:getBales()) do
-        local isValid, wrongWrapType = BaleToCollect.isValidBale(object, 
+        local isValid, wrongWrapType = BaleToCollect.isValidBale(object,
                 self.baleWrapper, self.baleLoader, self.baleWrapType)
         if isValid and g_baleToCollectManager:isValidBale(object) then
             local bale = BaleToCollect(object)
             -- if the bale has a mountObject it is already on the loader so ignore it
-            if not object.mountObject and object:getOwnerFarmId() == self.vehicle:getOwnerFarmId() and
-                    self:isBaleOnField(bale) then
+            if not object.mountObject and
+                    (object:getOwnerFarmId() == AccessHandler.EVERYONE or
```

Comment:
Maybe we could use the following function to check the access?
Should also check the _AccessHandler.EVERYONE_ permission. `g_currentMission.accessHandler:canFarmAccess(self.vehicle:getOwnerFarmId(), object)`